documentation/writing_tests.md

```diff
@@ -14,6 +20,139 @@
+
+New projects inside of this repository can take a safe guess based on how much the tests on average take locally. **Existing projects** should instead...
+
+    TODO: ASK ENGINEERING HOW TO GET THE MAX TIME PER TESTS
+
+#### On the usage of before, beforeEach, after and afterEach
+
+Mocha's `before` and `beforeEach` are methods that allow you to specify functions that should be executed either before all of the tests run or before each test runs, respectively. Similarly, `after` and `afterEach` exist so that developers can specify functions that should be executed after all tests run or after each test runs, respectively.
+
+We recommend using `beforeEach` rather than `before`, just as much as we recommend using `afterEach` rather than `after`. The idea is that each one of your test cases should not depend on a state that is shared with other tests. Use `beforeEach` to execute tasks that will prepare the resources needed for each test to run cleanly, and `afterEach` to tear down or clean those settings before the next test runs.
+
+We typically use `beforeEach` and `afterEach` to set up and tear down our test recorder. You can learn more about it in the section: [The Recorder](#the-recorder).
+
+#### Other general recommendations
+
+Mocha has many interesting features. Here's a list of our general recommendations on how to use some of these features:
+
+- **On describes:**  
+  Mocha's `describe` allows you to group test cases, even in nested groups. Take advantage of this. Group tests not only by the file that contains them, but also by what they're testing within that separation.
+- **On async:**  
+  Most of our test cases are asynchronous. Mocha allows you to write async test cases by calling `it` with an async function. Take this to your advantage. Use async functions on your test cases as much as possible. If all of your test cases use asynchronous functions, it will make them look and behave more consistently.
+- **On arrow function expressions:**  
+  Even though Mocha lets you write tests with arrow function expressions, as in `it("my test", async () => {})`, we recommend to use standard functions, as in `it("my test", async function () {})`, because standard functions have bindings to the execution context through the `this` keyword. Mocha's execution context is **necessary** to use the recorder, and it allows you to obtain the test name and other information that can be useful for your test cases. Mocha also discourages the use of arrow functions, which can be seen here: https://mochajs.org/#arrow-functions
+
+### Chai
+
+### Rollup
+
+### Karma
+
+### The Recorder
+
+## package.json scripts
+
+Your package.json test scripts related to mocha should end up looking like this:
```

Comment:
is it worth freezing these commands in a doc or should we link off to the template package.json?

Reply:
Linking to the template package.json is the way to go! This is heavily under development. Keep that in mind. Though, your feedback has been great so far! I really appreciate it. I will make sure to link to the template package.json 🌞

Reply:
I have added references to the template repo!